domain/src/main/java/woowacourse/movie/reservation/Reservation.kt

```diff
@@ -0,0 +1,17 @@
+package woowacourse.movie.reservation
+
+import woowacourse.movie.PaymentType
+import woowacourse.movie.bookingHistory.BookingDatabase
+import woowacourse.movie.ticket.Ticket
+import java.time.LocalDateTime
+
+data class Reservation(
+    val tickets: Set<Ticket>,
+    val id: Long = BookingDatabase.getNewId(),
+    val paymentType: PaymentType = PaymentType.OFFLINE,
+) {
+    val payment: Int get() = tickets.sumOf { it.price }
+    val movieId: Long get() = tickets.first().movieId
+    val bookedDateTime: LocalDateTime get() = tickets.first().bookedDateTime
```

Comment:
```suggestion
    val payment = tickets.sumOf { it.price }
```

tickets 는 재할당될 수 없는 프로퍼티 이므로 값이 바뀌지 않을꺼라, 바로 선언해주면 되지 않을까요?
getter 를 오버라이딩 할 경우, 값을 읽을때마다 계산이 진행될텐데 매번 같은 값이 전달될거라 낭비가 될 수 있을 것 같아요.